# **Week 2**
### Josh Winnes
### Topics in Data Science
### September 10, 2024

---
---

## **Data Setup**

In [1]:
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sns
import os

import dhs_util
from dhs_util import * 

os.chdir('/Users/joshwinnes/Library/Mobile Documents/com~apple~CloudDocs/Wheaton College/fall 2024/topics in data science/data')
df = pd.read_csv('dhs_service_records_synthesized_final.csv')

df = dhs_preprocessing(df)
df, service_map = add_service_label(df)
df = add_age_bin(df)

recipient = get_recipient_attribute(df)

/Users/joshwinnes/Library/Mobile Documents/com~apple~CloudDocs/Wheaton College/fall 2024/topics in data science/Math485/dhs_util.py:42: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  recipient = df.groupby(df['id'].cat.remove_unused_categories()).agg(


In [2]:
df.head()

id       date                                            service  age  \
0  1626 2021-01-31  Children_Attending_Early_Childhood_Programs_Ma...    5   
1  1626 2021-02-28  Children_Attending_Early_Childhood_Programs_Ma...    5   
2  1626 2021-03-31  Children_Attending_Early_Childhood_Programs_Ma...    5   
3  1626 2021-04-30  Children_Attending_Early_Childhood_Programs_Ma...    5   
4  1626 2021-05-31  Children_Attending_Early_Childhood_Programs_Ma...    5   

   gender     race   ethnicity     marital                      education  \
0  1~Male  1~White  99~Unknown  99~Unknown  9-12~High School (grade 9-12)   
1  1~Male  1~White  99~Unknown  99~Unknown  9-12~High School (grade 9-12)   
2  1~Male  1~White  99~Unknown  99~Unknown  9-12~High School (grade 9-12)   
3  1~Male  1~White  99~Unknown  99~Unknown  9-12~High School (grade 9-12)   
4  1~Male  1~White  99~Unknown  99~Unknown  9-12~High School (grade 9-12)   

      month serv age_bin  
0   January  S01     0-9  
1  February  S01     0-9  
2     March  S01     0-9  
3     April  S01     0-9  
4       May  S01     0-9

In [3]:
recipient.head()

id  num_service  distinct_service first_date  last_date  num_month  \
0   1           12                 1 2021-01-31 2021-12-31         12   
12  2            1                 1 2021-02-28 2021-02-28          1   
13  3           11                 1 2021-02-28 2021-12-31         11   
24  4            4                 1 2021-03-31 2021-12-31          4   
28  5           12                 1 2021-01-31 2021-12-31         12   

    distinct_month  age    gender                      race  \
0               12   13    1~Male                99~Unknown   
12               1   70  2~Female                   1~White   
13              11   29  2~Female                   1~White   
24               4    4  2~Female                   1~White   
28              12   40    1~Male  2~Black/African American   

                ethnicity                 marital  \
0   2~Not Hispanic/Latinx              99~Unknown   
12  2~Not Hispanic/Latinx               2~Married   
13  2~Not Hispanic/Latinx               2~Married   
24  2~Not Hispanic/Latinx  1~Single-Never Married   
28  2~Not Hispanic/Latinx               2~Married   

                                education serv age_bin  
0   GED~High School Diploma/GED Completed  S12   10-19  
12                             99~Unknown  S12   70-79  
13          9-12~High School (grade 9-12)  S12   20-29  
24          9-12~High School (grade 9-12)  S12     0-9  
28                 UNDERGRAD~Some College  S12   40-49

---
---

## **In Class Exercises**

### ***Tuesday, September 10, 2024***

### Exercise 1: How many recipients receive services every month in the year? 

In [5]:
recipient.query("distinct_month == 12 and num_service > 0").count()

id                  421900
num_service         421900
distinct_service    421900
first_date          421900
last_date           421900
num_month           421900
distinct_month      421900
age                 421900
gender              421900
race                421900
ethnicity           421900
marital             421900
education           421900
serv                421900
age_bin             421900
dtype: int64

We see that **421,900** recipients receive at least 1 service in every month.

### Exercise 2: What is the average number of services a recipient use? 

In [6]:
recipient['num_service'].mean()

13.331111523251261

Recipients use, on average, **13.331** services

### Exercise 3: If we remove the recipients who uses only one service in the year, what are the answers to the above questions?

In [11]:
recipient_more_than_one = recipient.query("num_service != 1").reset_index()
recipient_more_than_one.head()

index id  num_service  distinct_service first_date  last_date  num_month  \
0      0  1           12                 1 2021-01-31 2021-12-31         12   
1     13  3           11                 1 2021-02-28 2021-12-31         11   
2     24  4            4                 1 2021-03-31 2021-12-31          4   
3     28  5           12                 1 2021-01-31 2021-12-31         12   
4     40  6           24                 3 2021-01-31 2021-12-31         24   

   distinct_month  age    gender                      race  \
0              12   13    1~Male                99~Unknown   
1              11   29  2~Female                   1~White   
2               4    4  2~Female                   1~White   
3              12   40    1~Male  2~Black/African American   
4              12   71  2~Female  2~Black/African American   

               ethnicity                 marital  \
0  2~Not Hispanic/Latinx              99~Unknown   
1  2~Not Hispanic/Latinx               2~Married   
2  2~Not Hispanic/Latinx  1~Single-Never Married   
3  2~Not Hispanic/Latinx               2~Married   
4  2~Not Hispanic/Latinx  1~Single-Never Married   

                               education serv age_bin  
0  GED~High School Diploma/GED Completed  S12   10-19  
1          9-12~High School (grade 9-12)  S12   20-29  
2          9-12~High School (grade 9-12)  S12     0-9  
3                 UNDERGRAD~Some College  S12   40-49  
4                             99~Unknown  S09   70-79

In [12]:
recipient_more_than_one.query("distinct_month == 12 and num_service > 0").count()

index               421900
id                  421900
num_service         421900
distinct_service    421900
first_date          421900
last_date           421900
num_month           421900
distinct_month      421900
age                 421900
gender              421900
race                421900
ethnicity           421900
marital             421900
education           421900
serv                421900
age_bin             421900
dtype: int64

There are still **421,900** recipients who receive services every month, which makes sense because if they receive service every month, then they would have more than 1 service in the year and would not have been dropped.

In [13]:
recipient_more_than_one['num_service'].mean()

13.631203741194396

The mean jumped slightly to **13.631** services per recipient when we ignore the recipients who only received 1 service.

### Exercise 4: From all the plots, can we conclude that there is little difference in the number of services used between different age/gender/race groups?